assets/opengrep_rules/client/sensitive-data-access-validation.yaml

```diff
@@ -0,0 +1,50 @@
+rules:
+  - id: chromium-sensitive-data-access-validation
+    metadata:
+      author: Andrea Brancaleoni <abc@pompel.me>
+      references:
+        - https://github.com/brave/brave-browser/wiki/Security-reviews
+        - https://chromium.googlesource.com/chromium/src/+/main/docs/security/web-platform-security-guidelines.md
+      source: https://github.com/brave/security-action/blob/main/assets/opengrep_rules/client/sensitive-data-access-validation.yaml
+      assignees: |
+        thypon
+        fmarier
+        diracdeltas
+      category: security
+    languages: [cpp, c]
+    message: |
+      Access to sensitive user data (cookies, passwords, private browsing data)
+      requires security review. Ensure proper permission checks, incognito mode
+      isolation, and that sensitive data cannot leak across contexts.
+    severity: WARNING
+    patterns:
+      - patterns:
+          - pattern-either:
+              - pattern: GetCookies($URL, ...)
```

Comment:
as mentioned in the original PR these methods are not valid and will not capture what you are looking for here. Please see that comment for details.